components/connectivity-adapter/cmd/main.go

```diff
@@ -26,23 +26,31 @@ func main() {
 	err := envconfig.InitWithPrefix(&cfg, "APP")
 	exitOnError(err, "while loading app config")
 
+	handler, err := initAPIHandler(cfg)
+	if err != nil {
+		exitOnError(err, "Failed to init External Connector handler")
+	}
+
+	log.Printf("API listening on %s", cfg.APIAddress)
+	err = http.ListenAndServe(cfg.APIAddress, handler)
+	exitOnError(err, fmt.Sprintf("while listening on %s", cfg.APIAddress))
+}
+
+func initAPIHandler(cfg config) (http.Handler, error) {
 	router := mux.NewRouter()
 	router.HandleFunc("/v1/health", health.HandleFunc).Methods(http.MethodGet)
 
-	v1Router := router.PathPrefix("/{app-name}/v1").Subrouter()
+	applicationRegistryRouterV1 := router.PathPrefix("/{app-name}/v1").Subrouter()
+	connectorRouter := router.PathPrefix("/v1/applications").Subrouter()
 
-	appRegistryRouter := v1Router.PathPrefix("/metadata").Subrouter()
+	appRegistryRouter := applicationRegistryRouterV1.PathPrefix("/metadata").Subrouter()
 	appregistry.RegisterHandler(appRegistryRouter, cfg.AppRegistry)
-
-	connectorRouter := v1Router.PathPrefix("/applications").Subrouter()
-	err = connector.RegisterHandler(connectorRouter, cfg.Connector)
+	err := connector.RegisterHandler(connectorRouter, cfg.Connector, cfg.AppRegistry.DirectorEndpoint)
 	if err != nil {
-		exitOnError(err, "Failed to init Connector handler")
+		return nil, err
 	}
 
-	log.Printf("Listening on %s", cfg.Address)
-	err = http.ListenAndServe(cfg.Address, router)
-	exitOnError(err, fmt.Sprintf("while listening on %s", cfg.Address))
+	return connectorRouter, nil
```

Comment:
then `/v1/health` is not accessible anymore

Reply:
Done